Update criteria for .NET Foundation membership

BillWagner opened the pull request:
Prior to this year, the criteria was focused on contributions to .NET open source projects. That's only one possible way to participate.

This PR:

- clarifies what we look for when an open source maintainer or contributor requests to join.
- Adds criteria for those prospective members who identify more closely with other membership activities.

jguadagno commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+
+First, we validate the personal information:
+
+- Reasonable first name, last name.
```
Reasonable might be questioned. Instead

First Name and Last Name are reviewed to eliminate potential 'bots'.

jguadagno commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+First, we validate the personal information:
+
+- Reasonable first name, last name.
+- valid email address: We only validate that it is a valid email address format. Later, when an invitation is sent, that verifies that it's a real address.
```
valid email address: We only validate that it is a valid email address format. Later, when an invitation is sent, that verifies that it's a real address.
should be
A valid email address: *We only validate that it is a valid email address format. Later, when an invitation is sent, that verifies that it's a real address.*

jguadagno commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+
+We look to the educator's creation and delivery of curriculum in any of the following ways:
+
+- Teaching bootcamp courses, college or high school level courses in software development, where the .NET platform is a signification part of the curriculum.
```
Teaching. Whether that be Bootcamp courses, college, or high school level courses in software development where the .NET platform is a signification part of the curriculum.

jguadagno commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+
+See [Sharer of knowledge particpation type](membership-participation-types.md#the-sharer-of-knowledge).
+
+We look at the content they create and deliver to spread knowledge. These include:
```
The content that the applicant either creates and/or delivers to spread knowledge. These include, but are not limited to:

jguadagno commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+We look at the content they create and deliver to spread knowledge. These include:
+
+- books
+- conference talks.
```
No need for period

jguadagno commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+- books
+- conference talks.
+- twitch streams
+- blogs.
```
No need for period

jguadagno commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+- conference talks.
+- twitch streams
+- blogs.
+- online courseware.
```
No need for period

jguadagno commented on meetings/membership-criteria.md:
No need for period

jguadagno reviewed (changes requested):
I made a few suggestions

wordshaker commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+
+See [Open source maintainer](membership-participation-types.md#the-open-source-maintainer) and the [Open source contributor](membership-participation-types.md#the-open-source-contributor).
+
+We look at the contribution links. All activity must conform to the .NET Foundation CoC. Activities examined include:
```
Would be good to link it as well

CalvinAllen reviewed (approved):
Approved, though I agree with Joe's change requests

ovebastiansen commented:
Together with Joe's comments this looks good to me

CalvinAllen commented:
Went ahead and made those changes to save @BillWagner some work :)

BillWagner commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+
+- Events listed have a CoC that reflects the .NET Foundation's community guidelines.
+- Consistent track record holding events (virtual or in person).
+- Meetup is listed with the .NET Foundation
```
Clarify that this bullet only applies to meetup groups. (Conference organizers need not use meetup)

wordshaker replied:
@ovebastiansen pointed out the meetup being listed with the .NET Foundation is restrictive

BillWagner commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+
+We look at the event or user group activities. Examples include:
+
+- Events listed have a CoC that reflects the .NET Foundation's community guidelines.
```
Add link

wordshaker replied:
Add link to the community guidelines?

WhiteBlackGoose replied:
```suggestion
- Events listed have a CoC that reflects the .NET Foundation's [community guidelines](https://dotnetfoundation.org/about/code-of-conduct).
```

BillWagner commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+
+We look to the educator's creation and delivery of curriculum in any of the following ways:
+
+- Teaching. Whether that be bootcamp courses, college or high school level courses in software development where the .NET platform is a signification part of the curriculum.
```
Add online platforms (MS Learn, Pluralsight)

BillWagner commented on meetings/membership-criteria.md:
Point out that these (except maybe books) should show a track record of answers over time)

wordshaker commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+
+## Knowledge sharers
+
+See [Sharer of knowledge particpation type](membership-participation-types.md#the-sharer-of-knowledge).
```
particpation

WhiteBlackGoose replied:
```suggestion
See [Sharer of knowledge participation type](membership-participation-types.md#the-sharer-of-knowledge).
```

wordshaker commented on meetings/membership-criteria.md:
particpation

BillWagner commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+# Membership criteria
+
+When you apply for membership, we review your application based on the [membership participation types](membership-participation-types.md) defined. We ask for some basic information about you. 
```
Fix these links (different folder), etc)

WhiteBlackGoose replied:
```suggestion
When you apply for membership, we review your application based on the [membership participation types](../membership-participation-types.md) defined. We ask for some basic information about you. 
```

BillWagner commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+
+- First and Last name are reviewed to eliminate potential 'bots'.
+- A valid email address: We only validate that it is a valid email address format. Later, when an invitation is sent, that verifies that it's a real address.
+- A valid GitHub ID: We validate the profile appears to be human. There should be some activity. The user name must match any contribution links provided. GitHub activity mentioned in the contribution statement should match this ID.
```
This requirement can be removed. With the new portal, we don't need a GitHub id to communicate with members.

BillWagner commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+- A valid email address: We only validate that it is a valid email address format. Later, when an invitation is sent, that verifies that it's a real address.
+- A valid GitHub ID: We validate the profile appears to be human. There should be some activity. The user name must match any contribution links provided. GitHub activity mentioned in the contribution statement should match this ID.
+
+## Open Source Maintainer, Open Source Contributor
```
Question for the project committee. Could projects be hosted on another open platform? e.g. GitLab, BitBucket.

tannergooding replied:
Shaun should comment on this one. I don't think it's come up yet, however.

BillWagner commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+
+When you apply for membership, we review your application based on the [membership participation types](membership-participation-types.md) defined. We ask for some basic information about you. 
+
+## Basic information
```
In this section, make it clear that applicants can (but are not required) to fill in activities in multiple sections.

WhiteBlackGoose commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+
+We look at the contribution links. All activity must conform to the .NET Foundation [Code of Conduct (CoC)](https://dotnetfoundation.org/about/code-of-conduct). Activities examined include:
+
+- PRs created. We evaluate the participation in the community. Does the PR match the project's guidelines? Does the author engage constructively with reviewers?
```
```suggestion
- Pull Requests (PRs) created. We evaluate the participation in the community. Does the PR match the project's guidelines? Does the author engage constructively with reviewers?
```

WhiteBlackGoose commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+
+## Event organizer
+
+See [Event organizer particpation type](membership-participation-types.md#the-event-organizer).
```
```suggestion
See [Event organizer participation type](membership-participation-types.md#the-event-organizer).
```

WhiteBlackGoose commented on meetings/membership-criteria.md:
```diff
@@ -0,0 +1,64 @@
+
+## Educator
+
+See [Educator particpation type](membership-participation-types.md#the-educator).
```
```suggestion
See [Educator participation type](membership-participation-types.md#the-educator).
```

WhiteBlackGoose commented on meetings/membership-criteria.md:
```suggestion
See [Student participation type](membership-participation-types.md#the-student).
```

WhiteBlackGoose reviewed (commented):
You can add a recommendation to fix a line with suggestion (when reviewing a line, there's a button "add a suggestion"), if it makes it easier

BillWagner commented:
> You can add a recommendation to fix a line with suggestion (when reviewing a line, there's a button "add a suggestion"), if it makes it easier

We do usually do that. We were discussing live in our meeting yesterday and adding quick notes for our editing cycle. Thanks for adding all the suggestions. It save quite a bit of time.

BillWagner commented:
Thanks @jguadagno @CalvinAllen @wordshaker I've addressed all the comments from our meeting. Do you want to give it a last read?  I've also pinged the project committee to find out if we look at other OSS hosting options today.

CalvinAllen commented:
Ship it! 🚀
